packages/eslint-plugin/src/rules/key-spacing.ts

```diff
@@ -0,0 +1,404 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+
+const baseRule = getESLintCoreRule('key-spacing');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+const baseSchema = Array.isArray(baseRule.meta.schema)
+  ? baseRule.meta.schema[0]
+  : baseRule.meta.schema;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'key-spacing',
+  meta: {
+    type: 'layout',
+    docs: {
+      description:
+        'Enforce consistent spacing between property names and type annotations in types and interfaces',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: 'whitespace',
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: [baseSchema],
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [{}],
+
+  create(context, [options]) {
+    const sourceCode = context.getSourceCode();
+    const baseRules = baseRule.create(context);
+
+    /**
+     * Starting from the given a node (a property.key node here) looks forward
+     * until it finds the last token before a colon punctuator and returns it.
+     */
+    function getLastTokenBeforeColon(node: TSESTree.Node): TSESTree.Token {
+      const colonToken = sourceCode.getTokenAfter(node, util.isColonToken)!;
+
+      return sourceCode.getTokenBefore(colonToken)!;
+    }
+
+    /**
+     * Relevant nodes to our rule
+     *
+     * node.typeAnnotation will always be defined, but no way to enforce that and keep
+     * the type compatible with TSEstree.Node (except maybe TS 4.9' "satisfies" keyword)
```

Comment:
I added the test case in [`ae33b9b` (#6211)](https://github.com/typescript-eslint/typescript-eslint/pull/6211/commits/ae33b9b67c6a2a689c2bc3881ab2fc18b12c15fc):

```ts
let x: {
  a: number;
  abc: string;
}
```

Is that an `ImplicitAnyInterface`? Otherwise, can you give me a code sample for what you mean?